src/CLI.hs

```diff
@@ -261,38 +262,38 @@ runSMTWithTimeOut solver' maybeTimeout debug' sym
        runwithz3 = runSMTWith z3{verbose=debug'} $ (setTimeOut timeout) >> sym
 
 -- | Fail on error, or proceed with continuation
-proceed :: String -> Err a -> (a -> IO ()) -> IO ()
-proceed contents (Bad e) _ = prettyErr contents e
-proceed _ (Ok a) continue = continue a
-
-compile :: String -> Err [Claim]
-compile = pure . fmap annotate . enrich <=< typecheck <=< parse . lexer
-
-prettyErr :: String -> (Pn, String) -> IO ()
-prettyErr _ (pn, msg) | pn == nowhere = do
-  hPutStrLn stderr "Internal error:"
-  hPutStrLn stderr msg
-  exitFailure
-prettyErr contents (pn, msg) | pn == lastPos = do
-  let culprit = last $ lines contents
-      line' = length (lines contents) - 1
-      col  = length culprit
-  hPutStrLn stderr $ show line' <> " | " <> culprit
-  hPutStrLn stderr $ unpack (Text.replicate (col + length (show line' <> " | ") - 1) " " <> "^")
-  hPutStrLn stderr msg
-  exitFailure
-prettyErr contents (AlexPn _ line' col, msg) = do
-  let cxt = safeDrop (line' - 1) (lines contents)
-  hPutStrLn stderr $ show line' <> " | " <> head cxt
-  hPutStrLn stderr $ unpack (Text.replicate (col + length (show line' <> " | ") - 1) " " <> "^")
-  hPutStrLn stderr msg
-  exitFailure
+proceed :: Validate err => String -> err (NonEmpty (Pn, String)) a -> (a -> IO ()) -> IO ()
+proceed contents comp continue = validation (prettyErrs contents) continue (comp ^. revalidate)
+
+compile :: Bool -> String -> Error String [Claim]
```

Comment:
Correct! Maybe we should `enrich` after `annotate`, which would make it easier to run the whole compilation chain with/without `enrich`?

Reply:
I think the problem is not the order of operations, but rather that since `enrich` makes a bunch of implicit constraints explicit, it is not compatible with the round trips that we are fuzzing in the tests? I would be fine with just having the `compile` here always enrich and having a custom compile in the test code that just skips the enrich phase.

Reply:
Idk, I'm kind of not into having two separate functions which we need to ensure that they're consistent in order for things to work as expected. It's not about verbosity for me, it just seems like bad practice.

My point with the order of operations was that If we don't `enrich` in `compile`, we can just do `enrich <$> compile contents` in `CLI.hs` and `compile contents` in `test/Test.hs`.

I just pushed a change with that approach, lmk what you think.